pkg/templateservicebroker/api/types.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// +genclient=true
+
+type TemplateProvision struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	Spec   TemplateProvisionSpec
+	Status TemplateProvisionStatus
+}
+
+type TemplateProvisionSpec struct {
+	Template  templateapi.Template
+	SecretRef kapi.LocalObjectReference
+}
+
+type TemplateProvisionStatus struct {
+	State TemplateProvisionState
+}
+
+type TemplateProvisionState string
+
+const (
+	TemplateProvisionStateNew      TemplateProvisionState = "New"
+	TemplateProvisionStateCreating TemplateProvisionState = "Creating"
+	TemplateProvisionStateCreated  TemplateProvisionState = "Created"
+	TemplateProvisionStateFailed   TemplateProvisionState = "Failed"
+)
+
+// TemplateList is a list of Template objects.
+type TemplateProvisionList struct {
+	unversioned.TypeMeta
+	unversioned.ListMeta
+
+	Items []TemplateProvision
+}
+
+type UUIDMap struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	Spec UUIDMapSpec
+}
+
+type UUIDMapSpec struct {
+	TemplateProvisionRef kapi.ObjectReference
```

Comment:
I think the hope is that the use of this object goes away if the underlying API supports "contexts" (i.e. complex co-ordinates rather than just UUIDs) in the future.
My understanding from the call we had with @smarterclayton was that the same non-namespaced object kind used by the templateservicebroker shouldn't be reused elsewhere because it would blur the security boundaries between components.
But it would potentially be neat to have a generic mapping object, e.g.
Spec:
  From:
    Kind: TemplateBrokerUUID
    Value: <uuid>
  To:
    ObjectReference